contracts/ssnlist.scilla

```diff
@@ -1542,3 +1561,538 @@ transition DrainContractBalance(amt: Uint128, initiator: ByStr20)
+      | Some new_deleg_addr =>
+        lrc <- lastrewardcycle;
+        last_buf_deposit_cycle_deleg[new_deleg_addr][ssn_address] := lrc;
+        delete last_buf_deposit_cycle_deleg[initial_deleg][ssn_address];
+        e = { _eventname: "SwapLastBuffDepositCycleDeleg"; initial_deleg: initial_deleg; new_deleg: new_deleg_addr; ssn_addr: ssn_address; last_reward_cycle: lrc };
+        event e
+      | None => (* no new deleg to transfer to *)
+      end
+    | None => (* No last buff deposit cycle for this ssn *)
+    end
+  | None =>
+    e = DelegDoesNotExistAtSSN;
+    ThrowError e
+  end
+end
+
+(* @dev Swap deleg address from current_deleg -> new_deleg in withdrawal_pending *)
+procedure SwapPendingWithdrawal(blk_reward_cycle: Pair BNum Uint128)
+  current_deleg_o <- current_deleg;
+  new_deleg_o <- new_deleg;
+  fst = @fst BNum Uint128;
+  snd = @snd BNum Uint128;
+  blk_num = fst blk_reward_cycle;
+  reward = snd blk_reward_cycle;
+  
+  match new_deleg_o with
+  | Some new_deleg_addr =>
+    new_deleg_pending_withdrawal <- withdrawal_pending[new_deleg_addr][blk_num];
+    match new_deleg_pending_withdrawal with
+    | Some existing_reward =>
+      (* new deleg has exisiting blk_num *)
+      new_reward = builtin add existing_reward reward;
+      withdrawal_pending[new_deleg_addr][blk_num] := new_reward;
+      e = { _eventname: "SwapPendingWithdrawalExists"; initial_deleg: current_deleg_o; new_deleg: new_deleg_addr; existing_reward: existing_reward; new_reward: new_reward };
+      event e
+    | None =>
+      (* new deleg has no blk_num entry*)
+      withdrawal_pending[new_deleg_addr][blk_num] := reward;
+      e = { _eventname: "SwapPendingWithdrawalNewEntry"; initial_deleg: current_deleg_o; new_deleg: new_deleg_addr; reward: reward };
+      event e
+    end
+  | None => (* no new deleg to transfer to *)
+  end
+end
+
+
+(* @dev : Change a delegator's stake from initiator -> new_deleg_addr. *)
+procedure SwapDelegator(requestor: ByStr20, new_deleg_addr: ByStr20)
+  c = Some {ByStr20} requestor;
+  current_deleg := c;
+  n = Some {ByStr20} new_deleg_addr;
+  new_deleg := n;
+  ssnlist_o <- ssnlist;
+  ssninfo_list = builtin to_list ssnlist_o;
+  forall ssninfo_list SwapBuffDeposit;
+  forall ssninfo_list SwapDirectDeposit;
+  forall ssninfo_list SwapDelegStakePerCycle;
+  forall ssninfo_list SwapDepositAmtDeleg;
+  forall ssninfo_list SwapSSNDelegAmt;
+  forall ssninfo_list SwapLastBuffDepositCycleDeleg;
+  forall ssninfo_list SwapLastWithdrawCycleDeleg;
+  
+  (* swap pending withdrawal *)
+  match c with
+  | Some initial_deleg =>
+    deleg_blk_map <- withdrawal_pending[initial_deleg];
+    match deleg_blk_map with
+    | Some deleg_blk_entry =>
+      cycle_reward_list = builtin to_list deleg_blk_entry;
+      forall cycle_reward_list SwapPendingWithdrawal;
+      delete withdrawal_pending[initial_deleg]
+    | None => (* no pending withdrawal *)
+    end
+  | None =>
+    e = DelegDoesNotExistAtSSN;
```

Comment:
This error does not look accurate as pending withdrawal is no longer tie to any ssn